tools/kagami/src/genesis.rs

```diff
@@ -241,10 +259,231 @@ fn generate_synthetic(
                     ),
                 )
                 .into();
-                first_transaction.append_instruction(mint);
+                first_transaction.push_instruction(mint);
             }
         }
     }
 
     Ok(genesis)
 }
+
+/// Use `Kagami` to sign genesis block.
+#[derive(Clone, Debug, Parser)]
+#[command(group = ArgGroup::new("private_key").required(true))]
+#[command(group = ArgGroup::new("public_key").required(true))]
+pub struct SignArgs {
+    /// Path to genesis json file
+    #[clap(long)]
+    genesis_file: PathBuf,
+    /// The algorithm of the provided keypair
+    #[clap(default_value_t, long, short)]
+    algorithm: crypto::AlgorithmArg,
```

Comment:
The algorithm is included into the keypair itself. It is contained in both `PublicKey` and `PrivateKey`.